Tests/Unit/HttplugFactoryTest.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Http\HttplugBundle\Tests\Unit;
+
+use Http\HttplugBundle\ClientFactory\DummyClient;
+use Http\HttplugBundle\Collector\MessageJournal;
+use Http\HttplugBundle\HttplugFactory;
+use Puli\Discovery\Api\Discovery;
+use Puli\Discovery\Binding\ClassBinding;
+use Webmozart\Expression\Expr;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class HttplugFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var \Prophecy\Prophet
+     */
+    private $prophet;
```

Comment:
You don't need to manually instantiate a prophet object. Try `$this->prophesize()`